golos.publication/golos.publication.cpp

```diff
@@ -606,16 +611,22 @@ void publication::set_vote(name voter, const structures::mssgid& message_id, int
 
     auto sumcuratorsw_delta = get_delta(machine, FP(mssg_itr->state.voteshares), FP(msg_new_state.voteshares), pool->rules.curationfunc);
     msg_new_state.sumcuratorsw = (FP(mssg_itr->state.sumcuratorsw) + sumcuratorsw_delta).data();
+
+    fixp_t sharesfn = get_calc_sharesfn(
+                pool->rules.mainfunc.code,
+                FP(msg_new_state.netshares),
+                FP(pool->rules.mainfunc.maxarg)
+                );
+    wdfp_t total_rsharesfn = WP(pool->state.rsharesfn);
+    narrow_down(sharesfn, total_rsharesfn);
+
     message_index.modify(mssg_itr, _self, [&](auto &item) {
         item.state = msg_new_state;
         send_poststate_event(
             message_id.author,
             item,
-            get_calc_sharesfn(
-                pool->rules.mainfunc.code, 
-                FP(msg_new_state.netshares),
-                FP(pool->rules.mainfunc.maxarg)
-            )
+            static_cast<base_t>(elap_t(sharesfn).data()),
```

Comment:
It's prefer to pass `rsharesfn` to send_poststate_event in internal representation and translate into correct representation inside of `send_poststate_event`. So, all logic about change representation will be in one place.